lib/libzpool/kernel.c

```diff
@@ -497,8 +497,6 @@ vn_open(char *path, int x1, int flags, int mode, vnode_t **vpp, int x2, int x3)
 #ifdef __linux__
 		flags |= O_DIRECT;
 #endif
-		/* We shouldn't be writing to block devices in userspace */
-		VERIFY(!(flags & FWRITE));
```

Comment:
This was an additional sanity check which was added a while back when we only expected `zdb` to access the block devices read-only from user space.  When `zhack` was added that changed and this check wasn't dropped at the time.  One of the new test cases uncovered this since it uses `zhack` on a pool with disks instead of files.

Reply:
Makes sense. Thanks!